packages/monaco-graphql-editor/src/editor/OperationsEditor.tsx

```diff
@@ -0,0 +1,187 @@
+    insertText: entry.insertText ?? (entry.label as string),
+    insertTextRules: entry.insertText
+      ? languages.CompletionItemInsertTextRule.InsertAsSnippet
+      : undefined,
+    sortText: entry.sortText,
+    filterText: entry.filterText,
+    documentation: entry.documentation,
+    detail: entry.detail,
+    command: entry.command,
+  };
+};
+
+class GraphQLWorker {
+  doComplete(
+    documentModel: monaco.editor.IReadOnlyModel,
+    position: monaco.Position,
+    schema: GraphQLSchema
+  ): GraphQLWorkerCompletionItem[] {
+    const document = documentModel.getValue();
+    if (!document) {
+      console.log('no document');
+      return [];
+    }
+    if (!schema) {
+      console.log('no schema');
+    }
+    const graphQLPosition = toGraphQLPosition(position);
+    const suggestions = getAutocompleteSuggestions(
+      schema,
+      document,
+      graphQLPosition
+    );
+    return suggestions.map((suggestion) => this.toCompletion(suggestion));
+  }
+
+  toCompletion(
+    entry: CompletionItem,
+    range?: IRange
+  ): GraphQLWorkerCompletionItem {
+    return {
+      label: entry.label,
+      insertText: entry.insertText,
+      insertTextFormat: entry.insertTextFormat,
+      sortText: entry.sortText,
+      filterText: entry.filterText,
+      documentation: entry.documentation,
+      detail: entry.detail,
+      range: range ? toMonacoRange(range) : undefined,
+      kind: entry.kind,
+      command: entry.command
+        ? { ...entry.command, id: entry.command.command }
+        : undefined,
+    };
+  }
+}
+
+export const OperationsEditor: FC<{
+  schema: GraphQLSchema;
+  operations: string;
+}> = ({ schema, ...props }) => {
+  const monaco = useMonaco();
+  const completionProviderRef = useRef<monaco.IDisposable | null>(null);
+
+  useEffect(() => {
+    if (!monaco || !schema) {
+      return;
+    }
+    if (completionProviderRef.current) {
+      // dispose instance to prevent having duplicated field
+      completionProviderRef.current.dispose();
+    }
+    completionProviderRef.current =
+      monaco.languages.registerCompletionItemProvider('graphql', {
+        triggerCharacters: [':', '$', '\n', ' ', '(', '@'],
+        provideCompletionItems(
+          model: monaco.editor.IReadOnlyModel,
+          position: monaco.Position
+        ): monaco.languages.CompletionList {
+          try {
+            const worker = new GraphQLWorker();
+            const completionItems = worker.doComplete(model, position, schema);
+            return {
+              incomplete: true,
+              suggestions: completionItems.map((item) => toCompletion(item)),
+            };
+          } catch (err) {
+            console.error('Error fetching completion items', err);
+            return { suggestions: [] };
+          }
+        },
+      });
+  }, [monaco, schema]);
+
+  return (
+    <MonacoEditor
+      height="70vh"
+      {...props}
+      language="graphql"
+      defaultValue={props.operations}
```

Comment:
Cant, we just use props.defaultValue? No need to overwrite the property name.

Reply:
fixed